From 2ea81f956140f83355cab20a51b1a8c49d885cde Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Tue, 31 Oct 2023 16:28:09 +0000
Subject: [PATCH] libstdc++: replace deprecated std::is_pod<T> in //gpu

std::is_pod is deprecated since C++20. Replace with std::trivial and
std::is_standard_layout. Avoids a lot of warnings.

Bug: 957519
Change-Id: I8f5cb8b9d7db564a3ee9025897d8a8ac4602092f
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4994246
Reviewed-by: Austin Eng <enga@chromium.org>
Commit-Queue: Austin Eng <enga@chromium.org>
Auto-Submit: Stephan Hartmann <stha09@googlemail.com>
Cr-Commit-Position: refs/heads/main@{#1217649}
---
 gpu/command_buffer/service/dawn_platform.cc | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/gpu/command_buffer/service/dawn_platform.cc b/gpu/command_buffer/service/dawn_platform.cc
index f095e62759293b..e792075337da53 100644
--- a/gpu/command_buffer/service/dawn_platform.cc
+++ b/gpu/command_buffer/service/dawn_platform.cc
@@ -127,8 +127,10 @@ uint64_t DawnPlatform::AddTraceEvent(
   uint64_t result = 0;
   static_assert(sizeof(base::trace_event::TraceEventHandle) <= sizeof(result),
                 "TraceEventHandle must be at most the size of uint64_t");
-  static_assert(std::is_pod<base::trace_event::TraceEventHandle>(),
-                "TraceEventHandle must be memcpy'able");
+  static_assert(
+      std::is_trivial_v<base::trace_event::TraceEventHandle> &&
+          std::is_standard_layout_v<base::trace_event::TraceEventHandle>,
+      "TraceEventHandle must be memcpy'able");
   memcpy(&result, &handle, sizeof(base::trace_event::TraceEventHandle));
   return result;
 }
